.github/workflows/rust-test.yml

```diff
@@ -43,47 +43,8 @@ jobs:
           token: ${{ secrets.GITHUB_TOKEN }}
           args: --all-features
 
-  test-no-run:
-    name: Compile tests
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-
-      ### INSTALL RUST ###
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
-        id: toolchain
-        with:
-          toolchain: stable
-          profile: minimal
-          override: true
-
-      ### BUILD CACHE ###
-      - name: Cache Cargo registry, target, index
-        if: matrix.build != 'macos'
```

Comment:
I can't say I understand the significance of the minor difference `if: matrix.build != 'macos'` that exists here vs. below.

Reply:
This was in the `rust-test.yml` > `Compile tests` before I started and it is technically invalid.
 https://github.com/apollographql/federation/blob/bdb117763240947b2910341240a48144e5518453/.github/workflows/rust-test.yml#L60-L72

There isn't a `matrix` in play within that step!  If GitHub Actions ran validations on these, I would have expected it to have flagged that, though since it likely always evaluated to null (or something) this step would have always run (though again, this particular job is _only_ configured to ever run on `ubuntu-latest`, which is another point of failure). 

Good eye on your part in calling this out though!